### PR TITLE
[FLINK-14692][table]support rename/alter table in tableEnvironment in both flink/blink planner

### PR DESCRIPTION
**Pull request by zjuwangg**
## What is the purpose of the change

Support alterTableStatement as following:
1. ALTER TABLE  [[catalogName.] dataBasesName].tableName RENAME TO newTableName
2. ALTER TABLE  [[catalogName.] dataBasesName].tableName SET  ( name=value [, name=value]*)

## Brief change log
- [de0b373](https://github.com/apache/flink/commit/de0b3736069c62629a17afb60190abaa251ba33e) support rename/alter table operation

## Verifying this change

This change added tests and can be verified as follows:

  - *SqlToOperationConverterTest#testAlterTable*
  - *CatalogTableITCase.scala#testAlterTable*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (yes )
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

**Comment by zjuwangg** [comment]
cc @bowenli86 @xuefuz @lirui-apache @KurtYoung to have a review.

**Comment by zjuwangg** [comment]
@flinkbot run travis

**Comment by zjuwangg** [comment]
@flinkbot run travis

File: flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/operations/SqlToOperationConverter.java [hunk 1]
```diff
@@ -179,6 +184,37 @@ private Operation convertDropTable(SqlDropTable sqlDropTable) {
...
+		if (sqlAlterTable.isRename()) {
+			UnresolvedIdentifier newUnresolvedIdentifier = UnresolvedIdentifier.of(sqlAlterTable.fullNewTableName());
+			newTableIdentifer = catalogManager.qualifyIdentifier(newUnresolvedIdentifier);
+		} else {
+			Optional<CatalogManager.TableLookupResult> optionalCatalogTable = catalogManager.getTable(tableIdentifier);
+			if (optionalCatalogTable.isPresent() && !optionalCatalogTable.get().isTemporary()) {
+				CatalogTable originalCatalogTable = (CatalogTable) optionalCatalogTable.get().getTable();
+				properties.putAll(originalCatalogTable.getProperties());
+				sqlAlterTable.getPropertyList().getList().forEach(p ->
+								properties.put(((SqlTableOption) p).getKeyString().toLowerCase(),
```
Comment (bowenli86) [inline flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/operations/SqlToOperationConverter.java]: do we force users to have lower case keys?

Comment (zjuwangg) [reply flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/operations/SqlToOperationConverter.java]: That's a very good question. Now the create table converter will lower the properties too. And I talked with @danny0405  offline and agreed to change the current behavior, so I create a issue to track [FLINK-15061](https://issues.apache.org/jira/browse/FLINK-15061). I'll fix this problem in another PR.

**Comment by KurtYoung** [comment]
Paste sql standard here for later usage:

```<alter table statement> ::=
  ALTER TABLE <table name> <alter table action>

<alter table action> ::=
    <add column definition>
  | <alter column definition>
  | <drop column definition>
  | <add table constraint definition>
  | <alter table constraint definition>
  | <drop table constraint definition>
  | <add table period definition>
  | <drop table period definition>
  | <add system versioning clause>
  | <drop system versioning clause>

<add column definition> ::=
  ADD [ COLUMN ] <column definition>
```

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/ddl/AlterTableOperation.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import java.util.Map;
+
+/**
+ * Operation to describe a ALTER TABLE statement.
+ */
+public class AlterTableOperation implements AlterOperation {
+	private final ObjectIdentifier tableIdentifier;
+	private final ObjectIdentifier newTableIdentifier;
+	private final CatalogTable catalogTable;
+	private boolean isRename;
```
Comment (KurtYoung) [inline flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/ddl/AlterTableOperation.java]: According to sql standard, we will have lots of variant altering operator.  I think should prepare for the future changes. We have 2 options here:
1. Use `Enum` here to indicate the actual alter operation, like `RenameTable`, `AddColumn` and so on. But with solution, we might need to put all fields required by different operations into this class.
2. Use class hierarchy to represent alter operations, like have dedicated `RenameTableOperation`, `AddColumnOperation` here. 

I'm more favor of option #2, WDYT?

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/ddl/AlterTableOperation.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	public CatalogTable getCatalogTable() {
+		return catalogTable;
+	}
+
+	public boolean isRename() {
+		return isRename;
+	}
+
+	@Override
+	public String asSummaryString() {
```
Comment (KurtYoung) [inline flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/ddl/AlterTableOperation.java]: we should make this closer to sql standard.

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/ddl/AlterTableOperation.java (hunk 2 above)
Comment (zjuwangg) [reply flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/ddl/AlterTableOperation.java]: +1 for #2. 
I'll have a refactor later.

Comment (zjuwangg) [reply flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/ddl/AlterTableOperation.java]: done.

**Comment by bowenli86** [comment]
LGTM, will merge if @KurtYoung doesn't have more concerns

**Pull request merged**